streams/src/main/java/org/apache/kafka/streams/kstream/internals/InternalStreamsBuilder.java

```diff
@@ -312,6 +312,7 @@ public void buildAndOptimizeTopology(final Properties props) {
                 graphNodePriorityQueue.offer(graphNode);
             }
         }
+        internalTopologyBuilder.validateCoPartition();
```

Comment:
One question, since we do verification in topology builder, is there any validation code in later stage that could be removed?

Reply:
I'm not sure if we can remove later stage validation code

@mjsax @lkokhreidze Can you give some advice?

Reply:
I don't think so. Because the later verification that covers the case if the user specified all partitions, is the same code that verify co-partitioning depending on the input topic partition numbers. And we still need that check anyway.